### PR TITLE
Import monero into thor

### PR DESCRIPTION
**Pull request by tcharding**
Draft while I port the new `testcontainers::Image` stuff. Also before un-drafting I'll locally merge this into the baldr branch and check it all works.

We currently have a crate `monero-harness-rs` that started out as a crate to spin up a docker container for testing against monero. It turns out that the RPC clients (especially the monero-wallet-rpc RPC client) is useful and needed in the current work on the Baldr project.

Import `monero-harnes-rs` into this workspace as `monero`. Slightly rearrange the directory tree to put the two RPC clients under an `rpc` submodule. This is to facilitate upstreaming of the `rpc` module into `monero-rs` once the API stabalises and it implements the whole RPC API (or a decent part of it).

There is one PR remaining on the `monero-harness-rs` crate, it implements `testcontainers::Image` instead of using the `GenericImage`. Has bugs, I will manually bring the changes across to here once it merges.

### CODE REVIEW

**Comment by thomaseizinger** [comment]
I am gonna remove my review assignment from this if you don't mind because I am not sure if my view adds value here (and the code is just a copy-paste I assume) :)

**Comment by tcharding** [comment]
No worries, I only added you because it was your suggestion to do this move, thought you might like to see I was listening to advice :)

File: monero/.github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+name: CI
```
Comment (bonomat) [inline monero/.github/workflows/ci.yml:1]: Is this file picked-up by Github CI? I think you will need to move it one folder up.

Comment (tcharding) [reply monero/.github/workflows/ci.yml]: oh that shouldn't be in there at all.  Thanks

**Pull request merged**